modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/introspection/describer/XmlBasedDescriber.java

```diff
@@ -0,0 +1,285 @@
+      .put("datetime", typeLoader.load(Calendar.class))
+      .put("date", typeLoader.load(Date.class))
+      .put("integer", typeLoader.load(Integer.class))
+      .put("time", typeLoader.load(LocalTime.class))
+      .build();
+
+  private static final ComponentIdentifier OPERATION_IDENTIFIER =
+      new ComponentIdentifier.Builder().withNamespace(MODULE_NAMESPACE_NAME).withName("operation").build();
+  private static final ComponentIdentifier OPERATION_PROPERTY_IDENTIFIER =
+      new ComponentIdentifier.Builder().withNamespace(MODULE_NAMESPACE_NAME).withName("property").build();
+  private static final ComponentIdentifier OPERATION_PARAMETERS_IDENTIFIER =
+      new ComponentIdentifier.Builder().withNamespace(MODULE_NAMESPACE_NAME).withName("parameters").build();
+  private static final ComponentIdentifier OPERATION_PARAMETER_IDENTIFIER =
+      new ComponentIdentifier.Builder().withNamespace(MODULE_NAMESPACE_NAME).withName("parameter").build();
+  private static final ComponentIdentifier OPERATION_BODY_IDENTIFIER =
+      new ComponentIdentifier.Builder().withNamespace(MODULE_NAMESPACE_NAME).withName("body").build();
+  private static final ComponentIdentifier OPERATION_OUTPUT_IDENTIFIER =
+      new ComponentIdentifier.Builder().withNamespace(MODULE_NAMESPACE_NAME).withName("output").build();
+  private static final ComponentIdentifier MODULE_IDENTIFIER =
+      new ComponentIdentifier.Builder().withNamespace(MODULE_NAMESPACE_NAME).withName(MODULE_NAMESPACE_NAME)
+          .build();
+
+  private final String modulePath;
+
+  /**
+   * @param modulePath relative path to a file that will be loaded from the current {@link ClassLoader}.
+   */
+  public XmlBasedDescriber(String modulePath) {
+    this.modulePath = modulePath;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public ExtensionDeclarer describe(DescribingContext context) {
+    // We will assume the current class loader will be the one defined for the plugin (which is not filtered and will allow us to access any resource in it
+    URL resource = currentThread().getContextClassLoader().getResource(modulePath);
+    if (resource == null) {
+      throw new IllegalArgumentException(format("There's no reachable XML in the path '%s'", modulePath));
+    }
+
+    XmlConfigurationDocumentLoader xmlConfigurationDocumentLoader = new XmlConfigurationDocumentLoader();
+
+    Document moduleDocument = getModuleDocument(xmlConfigurationDocumentLoader, resource);
+    XmlApplicationParser xmlApplicationParser = new XmlApplicationParser(new SpiServiceRegistry());
+    Optional<ConfigLine> parseModule = xmlApplicationParser.parse(moduleDocument.getDocumentElement());
+    if (!parseModule.isPresent()) {
+      //this happens in org.mule.runtime.config.spring.dsl.processor.xml.XmlApplicationParser.configLineFromElement()
+      throw new IllegalArgumentException(format("There was an issue trying to read the stream of '%s'", resource.getFile()));
```

Comment:
Is not the real problem that the descriptor is not well formed instead of being a problem reading the data?